scargo/commands/flash.py

```diff
@@ -5,28 +5,38 @@
 import subprocess
 import sys
 from pathlib import Path
+from typing import Optional
 
 from scargo.config import Config
 from scargo.config_utils import prepare_config
 from scargo.logger import get_logger
 from scargo.path_utils import get_project_root
 
 
-def scargo_flash(app: bool, fs: bool, flash_profile: str) -> None:
+def scargo_flash(
+    app: bool, fs: bool, flash_profile: str, port: Optional[str] = None
+) -> None:
     config = prepare_config()
     target = config.project.target
+    logger = get_logger()
 
+
+    if port and target.family != "esp32":
+        logger.warning("--port option is only supported for esp32 projects.")
     if target.family == "esp32":
-        flash_esp32(config, app=app, fs=fs, flash_profile=flash_profile)
+        flash_esp32(config, app=app, fs=fs, flash_profile=flash_profile, port=port)
     elif target.family == "stm32":
         flash_stm32(config, flash_profile)
     else:
-        logger = get_logger()
         logger.error("Flash command not supported for this target yet.")
 
 
 def flash_esp32(
-    config: Config, app: bool, fs: bool, flash_profile: str = "Debug"
+    config: Config,
+    app: bool,
+    fs: bool,
+    flash_profile: str = "Debug",
+    port: Optional[str]=None,
```

Comment:
This line would be reformetted by `black`. I recommend installing `pre-commit` to prevent similar problems in the future:
```bash
pip install pre-commit
pre-commit install
```

Reply:
`pre-commit` runs on modified (and staged) files before each commit, but you can also run it manually on the whole repository with `pre-commit run --all`